MLS.Agent/NativeAssemblyLoadHelper.cs

```diff
@@ -15,6 +15,7 @@ namespace MLS.Agent
 {
     public class NativeAssemblyLoadHelper : INativeAssemblyLoadHelper
     {
+        private static readonly HashSet<DirectoryInfo> globalProbingPaths = new HashSet<DirectoryInfo>();
```

Comment:
There are several instances of NativeAssemblyLoadHelper  created.  The globalProbingPaths  is the union of all of the located probing paths.